### PR TITLE
osd/PGLog: get rid of ineffective container operations

### PR DESCRIPTION
**Pull request by xiexingguo**
Signed-off-by: xie xingguo <xie.xingguo@zte.com.cn>

### CODE REVIEW

**Comment by xiexingguo** [comment]
OT:

@tchaikov Hi, kefu. Where can I get the OSD logs now? E.g.:

http://qa-proxy.ceph.com/teuthology/xxg-2017-11-27_05:37:53-rados-wip-xxg-testing-distro-basic-mira/1895723/

I have some problems need to consult the OSD-side logs... 
Thanks!

**Comment by tchaikov** [comment]
the test timed out after running for 12 hours. you need rerun the test, and to grab the log manually.

**Comment by xiexingguo** [comment]
> the test timed out after running for 12 hours. you need rerun the test, and to grab the log manually.

Thanks @tchaikov !

**Comment by tchaikov** [comment]
http://pulpito.ceph.com/kchai-2017-12-01_08:15:14-rados-wip-kefu-testing-2017-12-01-1256-distro-basic-mira/

File: src/osd/osd_types.h [hunk 1]
```diff
@@ -3835,6 +3835,14 @@ class pg_missing_set : public pg_missing_const_i {
       return false;
     return true;
   }
+  eversion_t get_oldest_need() {
```
Comment (tchaikov) [inline src/osd/osd_types.h]: @xiexingguo could you make this method `const`?

Comment (xiexingguo) [reply src/osd/osd_types.h]: @tchaikov Sure, done.

**Pull request merged**